spec/mail/messagedelivery.mdown

```diff
@@ -0,0 +1,221 @@
+
+### Response
+
+The response to *setMessageDeliveries* is called *messagesDeliveriesSet*. It has the following arguments:
+
+- **accountId**: `String`
+  The id of the account used for the call.
+- **oldState**: `String|null`
+  The state string that would have been returned by *getMessageDeliveries* before making the requested changes, or `null` if the server doesn't know what the previous state string was.
+- **newState**: `String`
+  The state string that will now be returned by *getMessageDeliveries*.
+- **created**: `String[Message]`
+  A map of the creation id to an object containing any properties on the successfully created MessageDelivery object that were not supplied by the client, or have been changed by the server from that supplied by the client.
+- **destroyed**: `String[]`
+  A list of ids for MessageDelivery objects that were successfully destroyed.
+- **notCreated**: `String[SetError]`
+  A map of creation id to a SetError object for each MessageDelivery that failed to be created. The possible errors are defined above.
+- **notDestroyed**: `String[SetError]`
+  A map of MessageDelivery id to a SetError object for each MessageDelivery that failed to be destroyed. The possible errors are defined above.
+
+The following errors may be returned instead of the *messagesSet* response:
+
+`accountNotFound`: Returned if an *accountId* was explicitly included with the request, but it does not correspond to a valid account.
+
+`accountNotSupportedByMethod`: Returned if the *accountId* given corresponds to a valid account, but the account does not support this data type.
+
+`accountReadOnly`: Returned if the account has `isReadOnly == true`.
+
+`invalidArguments`: Returned if one of the arguments is of the wrong type, or otherwise invalid. A *description* property MAY be present on the response object to help debug with an explanation of what the problem was.
+
+`stateMismatch`: Returned if an *ifInState* argument was supplied and it does not match the current state.
+
+### Implicit call to setMessages
+
+For each successful create/destroy, the server should see if there is an entry with the same id in the onCreateUpdateMessage/onDestroyUpdateMessage properties respectively. If so, add an entry to an (initially empty) *update* object, with the key being the *messageId* on the MessageDelivery object that was created/destroyed, and the value being the value from the onCreateUpdateMessage/onDestroyUpdateMessage object.
+
+After the *messagesDeliveriesSet* response is emitted, if the *update* object is not empty, an implicit call is made to *setMessages*, with the following arguments:
+
+- **accountId**: Same as for the call to *setMessageDeliveries*.
+- **update**: The *update* object as calculated above.
```

Comment:
You do not define what an update is. In my opinion, the knowledge of the required update should be explicitly embedded to MessageDelivery.